systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,333 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+
+        LOGGER.info("[TLS] Sending messages to topic {}, cluster {} and message count of {}",
+                topicSourceName, kafkaClusterSourceName, messagesCount);
+        InternalKafkaClient producerInternalKafkaClient = internalKafkaClient.toBuilder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(topicSourceName)
+                .withClusterName(kafkaClusterSourceName)
+                .withKafkaUsername(userTlsSource.getMetadata().getName())
+                .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                .build();
+        producerInternalKafkaClient.checkProducedAndConsumedMessages(producerInternalKafkaClient.sendMessagesTls(), 200);
+
+        LOGGER.info("[TLS] Wait 90s until MM2 refreshes its status.");
+        Thread.sleep(Duration.ofSeconds(90).toMillis());
+
+        LOGGER.info("[TLS] Receiving messages from topic {}, cluster {} and message count of {}",
+                topicTargetName, kafkaClusterTargetName, messagesCount);
+        InternalKafkaClient receiverInternalKafkaClient = producerInternalKafkaClient.toBuilder()
+                .withTopicName(topicTargetName)
+                .withClusterName(kafkaClusterTargetName)
+                .withKafkaUsername(userTlsTarget.getMetadata().getName())
+                .build();
+        receiverInternalKafkaClient.checkProducedAndConsumedMessages(200, receiverInternalKafkaClient.receiveMessagesTls());
+
+        LOGGER.info("[TLS] Going to delete ClientsCA certificates");
+        SecretUtils.deleteSecretWithWait(KafkaResources.clientsCaCertificateSecretName(kafkaClusterSourceName), namespaceName);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName), 3, kafkaSourcePods);
+        DeploymentUtils.waitForDeploymentAndPodsReady(KafkaResources.entityOperatorDeploymentName(kafkaClusterSourceName), 1);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        SecretUtils.deleteSecretWithWait(KafkaResources.clientsCaCertificateSecretName(kafkaClusterTargetName), namespaceName);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(kafkaClusterTargetName), 3, kafkaTargetPods);
+        DeploymentUtils.waitForDeploymentAndPodsReady(KafkaResources.entityOperatorDeploymentName(kafkaClusterTargetName), 1);
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("[TLS] Wait 120s until MM2 refreshes its status.");
+        Thread.sleep(Duration.ofSeconds(240).toMillis());
+        LOGGER.info("[TLS] Send and receive messages after clients certs were removed");
+        producerInternalKafkaClient.checkProducedAndConsumedMessages(producerInternalKafkaClient.sendMessagesTls(), receiverInternalKafkaClient.receiveMessagesTls());
```

Comment:
So now you are receiving and sending from/to target cluster?

Reply:
No I believe it's sending to Source and receiving from mirrored Target. 
But now I am not sure, so I will check it out tomorrow. :-)